main.go

```diff
@@ -17,6 +17,8 @@ import (
 var (
 	verbose = flag.Bool("verbose", false, "Print logs to stderr")
 	nodeURL = flag.String("node-addr", "/ip4/127.0.0.1/tcp/5001/", "IPFS node URL")
+	publish = flag.Bool("publish", false, "Option to publish to IPNS")
+	key     = flag.String("key", "", "key to publish to publish to IPNS")
```

Comment:
could it be one flag?

Reply:
Yeah sure. We can make it --publish=<IPNS-key>. And for self we can say --publish=""

Reply:
`--ipns-publish-key` or `--ipns-key` and `--ipns-key=self` for self

it makes it easier for new users to understand

Reply:
or `--publish-key`

Reply:
Done